src/cow-react/common/hooks/useTokenBySymbolOrAddress.ts

```diff
@@ -2,10 +2,15 @@ import { NativeCurrency, Token } from '@uniswap/sdk-core'
 import { useMemo } from 'react'
 import { useAllTokensList } from 'cow-react/common/hooks/useAllTokensList'
 import useNativeCurrency from 'lib/hooks/useNativeCurrency'
+import { useFavouriteTokens } from 'state/user/hooks'
 
 export function useTokenBySymbolOrAddress(symbolOrAddress?: string | null): Token | NativeCurrency | null {
   const tokens = useAllTokensList()
   const nativeCurrency = useNativeCurrency()
+  const favouriteTokens = useFavouriteTokens()
+  const checkBySymbolAndAddress = (token: Token, symbolOrAddress: string) =>
+    token.address.toLowerCase() === symbolOrAddress.toLowerCase() ||
+    token.symbol?.toLowerCase() === symbolOrAddress.toLowerCase()
```

Comment:
Can this function be moved outside the hook?

Reply:
Done